src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/on_app_interface_unregistered_notification_test.cc

```diff
@@ -0,0 +1,143 @@
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "mobile/on_app_interface_unregistered_notification.h"
+
+#include <utility>
+
+#include "gtest/gtest.h"
+
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/smart_object_keys.h"
+#include "interfaces/MOBILE_API.h"
+#include "smart_objects/smart_object.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace on_app_interface_unregistered_notification {
+
+namespace am = ::application_manager;
+
+using sdl_rpc_plugin::commands::OnAppInterfaceUnregisteredNotification;
+using Reason = mobile_apis::AppInterfaceUnregisteredReason::eType;
+using ::testing::Return;
+
+namespace {
+const uint32_t kAppId = 1u;
+}  // namespace
+
+class OnAppInterfaceUnregisteredNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  OnAppInterfaceUnregisteredNotificationTest()
+      : message_(CreateMessage()), command_(), mock_app_(CreateMockApp()) {
+    (*message_)[am::strings::params][am::strings::connection_key] = kAppId;
+    ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app_));
+    CommandsTest<CommandsTestMocks::kIsNice>::InitCommand(kDefaultTimeout_);
+    command_.reset(
+        new OnAppInterfaceUnregisteredNotification(message_,
+                                                   app_mngr_,
+                                                   mock_rpc_service_,
+                                                   mock_hmi_capabilities_,
+                                                   mock_policy_handler_));
+  }
+
+  am::commands::MessageSharedPtr message_;
+  std::unique_ptr<OnAppInterfaceUnregisteredNotification> command_;
+  MockAppPtr mock_app_;
+};
+
+const auto data_to_check = [](const smart_objects::SmartObject& so) {
```

Comment:
Not obvious name - instead of check this lambda contains some kind of conversion from `SmartObject` to `std::pair`.
Another point - lambda function is still a function, not a simple variable, so its naming should be appropriate.

Reply:
@YarikMamykin Done in cfb245dd9b7c5922c4b0ab05ef3f2f25f25fd88b